daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/BCDParsers.scala

```diff
@@ -34,9 +34,8 @@ class BCDDecimalKnownLengthParser(
 ) extends PackedBinaryDecimalBaseParser(e, binaryDecimalVirtualPoint)
   with HasKnownLengthInBits {
 
-  override def toBigInteger(num: Array[Byte]): JBigInteger = DecimalUtils.bcdToBigInteger(num)
-  override def toBigDecimal(num: Array[Byte], scale: Int): JBigDecimal =
-    DecimalUtils.bcdToBigDecimal(num, scale)
+  override def toNumber(num: Array[Byte]): JBigDecimal =
```

Comment:
Why change this to toNumber? The return type is not JNumber. 

Is there scaladoc on the toNumber method? There needs to be to explain this naming convention. I don't really understand why this got renamed.

Reply:
The trait these classes mix in requires an implementation of `def toNumber(num: Array[Byte]): A`, where `A <: Number`. We could remove the `A` type parameter and just make it so it always returns a `JNumber`, but then implementations have to be a bit more careful to return the right thing.

As it is right now, the type parameter ensures the implementations of `toNumber` return the right kind of `JNumber` (some should always return a BigInteger, some should always return a BigDecimal). I'm not sure the type parameter is used anywhere else though, so maybe that's overkill and just adds unnecessary complication?

Reply:
I think the type parameter is almost certainly the right thing, I just don't see a change that adds this toNumber super method declaration, yet it's not in the current Daffodil main branch, so when did this type parameter get added? 

But what is the purpose of the toNumber method? I think it is "convert to a number type we can store in the Infoset object. This should be the most specific meaningful subtype of JNumber" that is used in the Infoset object representation. Or something like that.

Reply:
The top level declaration of toNumber is in PackedBinaryTraits.scala where it is simply declared as
`trait PackedBinaryConversion[A <: Number] {
    def toNumber(num: Array[Byte]): A
    ....`

The actual implementations are in the classes that extend the PackedBinaryConversion trait, like BCDIntegerKnownLengthParser.

The function will only generate either BigInteger's or BigDecimal's depending on the calling classes type, so perhaps a better name for the function would be "toBigNumber".  The BigNumber then gets converted to the appropriate PrimType by calling PrimNumberic.fromNumber(bigNumber).